src/node.rs

```diff
@@ -107,12 +108,16 @@ impl<S: StorageExt> SawtoothRaftNode<S> {
         } {
             debug!("Leader({:?}) transition to Validating block {:?}", self.peer_id, block.block_id);
             self.leader_state = Some(LeaderState::Validating(block.block_id.clone()));
+            leader_state_publishing_this_block = true;
         }
 
-        debug!("Block has been received: {:?}", &block);
-        self.block_queue.block_new(block.clone());
-
-        self.service.check_blocks(vec![block.block_id]).expect("Failed to send check blocks");
+        // Add this block to block queue if it's a block which leader has published
+        // or if Raft node is in follower state
+        if leader_state_publishing_this_block || self.follower_state.is_some() {
+            debug!("Block has been received: {:?}", &block);
+            self.block_queue.block_new(block.clone());
```

Comment:
Do you think similar check is required before adding a new block to the commit_queue? "validator_backlog" is still ok to have extra blocks in it but commit_queue should not have extra blocks which are not to be committed. Example if leader has changed and the block arrives after validation.

Reply:
Do you mean in `on_block_valid()` check if the block arrived in follower/leader state is same as the block for which check_blocks was called earlier in `on_block_new()` while it was in corresponding follower/leader state. So for example, if we called `check_blocks()` for block 1 in leader state and we change to follower state and then block 1 arrived then ignore block 1 ?

Reply:
This can be handled in separate if required as is not related to this PR